### PR TITLE
[refactor](pipelineX) refine union dependency

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run pipelinex_p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/exec/union_source_operator.h [hunk 1]
```diff
@@ -79,13 +79,8 @@ class UnionSourceDependency final : public Dependency {
         if (((UnionSharedState*)_shared_state.get())->child_count() == 0) {
             return nullptr;
         }
-        if (((UnionSharedState*)_shared_state.get())->data_queue.is_all_finish() ||
-            ((UnionSharedState*)_shared_state.get())->data_queue.remaining_has_data()) {
-            return nullptr;
-        }
-        return this;
+        return Dependency::is_blocked_by(task);
```
Comment (Gabriel39) [inline be/src/pipeline/exec/union_source_operator.h:82]: Not to implement `is_blocked_by`

File: be/src/pipeline/exec/data_queue.h [hunk 2]
```diff
@@ -62,7 +62,13 @@ class DataQueue {
     bool data_exhausted() const { return _data_exhausted; }
     void set_dependency(Dependency* source_dependency, Dependency* sink_dependency) {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/data_queue.h:63]: delete this function

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run pipelinex_p0

**Review by Gabriel39 — APPROVED** [review]

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**